addons/product/models/res_partner.py

```diff
@@ -30,9 +30,17 @@ def _inverse_product_pricelist(self):
         )
         default_for_country = pls and pls[0]
         actual = self.env['ir.property'].get('property_product_pricelist', 'res.partner', 'res.partner,%s' % self.id)
-
+        # We want to keep the current pricelist if it's still valid for the new country
+        # or if the country would be the same, like in the case of an state change that
+        # triggers this method and could lead to loose the original pricelist
```

Comment:
```suggestion
        # triggers this method and could lead to lose the original pricelist
```

![](https://i.pinimg.com/564x/49/c4/f9/49c4f9f515ed6fb55def82884872508d.jpg)

Reply:
:sweat_smile: